More updates to the Elevation Profile view

DelazJ opened the pull request:
* Restructure the elevation profile chapter, fixing typos and duplicates and following this sequence (left: current, right: PR)
![image](https://github.com/qgis/QGIS-Documentation/assets/7983394/4cd86b9e-4212-4c55-a5b4-c05dfb9b6269)

* Move up the Elevation chapter in the index as a transitional chapter between 2D and 3D views
* Add connections to the other chapters that should reference the profile tool
* Report icons to the guidelines substitutions
<!---
Include a few sentences describing the overall goals for this Pull Request.
 
A list of issues is at https://github.com/qgis/QGIS-Documentation/issues.
Add "fix #issuenumber" for each issue the PR fixes. The ticket(s) will be closed automatically.
If your PR doesn't fix entirely the ticket, only add the ticket(s) reference preceded by # character.
-->
Goal: Fixes #7485, fixes #7500, fixes #7507, fixes #7523, fixes #7531, fixes #7544, fixes #7546

Ticket(s): #
<!---
Indicate whether the fix should be backported to previous release.
Replace the space between square brackets by a `x` to make it checked.
-->
- [x] Backport to LTR documentation is requested

<!---
Reviewing is a process done by community members, mostly on a volunteer basis.
We try to keep the overhead as small as possible and appreciate if you help us.
Please read carefully and ensure you comply with our writing guidelines at
https://docs.qgis.org/testing/en/docs/documentation_guidelines/index.html.
Feel free to ask in a comment or the (qgis-community-team mailing list)
[https://lists.osgeo.org/mailman/listinfo/qgis-community-team] if you have troubles with any item.
--->

diethard2 commented:
@DelazJ  Thanks for putting so much work into this! I have looked into the change and where I to tried to make the change more atomic and first tried to keep it as small as possible I see that you have included many changes in many places. Places that I would never have thought of. It has indeed become a big change.

To put the chapter of Elevation Profile View between 2D map view and 3D map view was a surprise. I would rather keep the original order which makes more sense if you read the contents of the manual. 

You have reorganized the elevation chapter a lot. I liked the structure of 3D Map View chapter and have build the first version on basis of that chapter, I tried to make it easy to read. I see you used a different more formal style i.e. for explaining the toolbar section. I have not rebuild the chapter yet, but I assume proposed changes makes it more formal but a bit less enjoyable to read. You have included some good new pointers like that when you close the window the elevation profile is lost. Actually when you close the project, the elevation panel will be closed and you will lose the changes as well. A save option (like the 3D Map View would be a nice to have.

You removed the troubleshoot section on how to create 3D layers for 2D layers using processing tool. The idea of including a troubleshoot chapter is maybe new, but I wonder what is wrong when you include answers to problems that many users will face when creating elevation profile?  Could you consider to include it for now and maybe discuss with others if we should include more troubleshoot sections in other places or keep the manual more focused on explaining interfaces.

I leave the decision on changes entirely up to you. I just wanted to reflect some thoughts.
Sorry I have missed some comments in the past. I can only work very occasionally in my free time on QGIS Documentation. Please do not feel frustrated when I did not react in time, I am not that focused on social media.
I am glad soon we will have a new chapter on the Elevation Profile View in 3.28! I will use it in the 'classroom' soon!
Regards, Diethard

DelazJ commented:
@diethard2 Thanks for your feedback.
Most of the changes included here were already mentioned in https://github.com/DelazJ/QGIS-Documentation/pull/99 that I sent an email to you for review, twice (because I'm not on social media, either). Unfortunately, without success. And yes, TBH, I felt frustrated. But I'm not anymore and I'm happy of your feedback. Really!
Please take the time to read my comments there to get the rationale of the changes and let's discuss here or there, as you want.
I'll just address a few points below.
> @DelazJ Thanks for putting so much work into this! I have looked into the change and where I to tried to make the change more atomic and first tried to keep it as small as possible I see that you have included many changes in many places. Places that I would never have thought of. It has indeed become a big change.

I can't tell if you appreciate or disagree 😃. If the former, then 👍🏿. If the latter, please add comments to particular lines you think they shouldn't be there. It will make it easier to rework. Note that there's almost no new feature described in the elevation_profile.rst, just moved around or expanded a bit what you described. 
Also note that the diff looks big because I split some of your long paragraphs into smaller lines, easier/faster to review/edit, and due to the rst indentation system for lists.
> To put the chapter of Elevation Profile View between 2D map view and 3D map view was a surprise. I would rather keep the original order which makes more sense if you read the contents of the manual.

Maybe I don't get the logic you want to keep (mine was about moving from a planar view to a side view with some elevation, and then to 3D view) but I do not have a strong opinion, so no issue with rolling back.
> I have not rebuild the chapter yet, but I assume proposed changes makes it more formal but a bit less enjoyable to read.

You don't need to rebuild yourself, it is available at https://github.com/qgis/QGIS-Documentation/actions/runs/5355207042
> You removed the troubleshoot section on how to create 3D layers for 2D layers using processing tool

Speaking specifically of this one, IMHO it is wrong to assume and advise that you need 3D layers to have the features displayed in the elevation view. Just configure a height/extrusion property in the elevation tab looks enough to display 2D vector data.
some `.. note::`, `.. hint::`, `.. tip::` and so on were meant to sometimes address issues people may face and have advantages that they can be included at any place in the page, i.e. closer to where the topic is being addressed. I don't know if we need dedicated section, but if so, we should ensure it is correct. Anyway, where would you see such a consultation?
> I leave the decision on changes entirely up to you.

With all due respect, NO, you shouldn't. QGIS documentation is not MY thing. As maintainers, we both (as many people that will get notification of this message) have the same rights and same responsibility. And if you disagree with some changes, then let's spend time to find what is more appropriate for the project, and not what best suits me. And this could start with adding suggestion to specific places.

I hope that none of my comments above hurts you. If so, sorry and it is clearly unintentional. Looking forward to your reply...

diethard2 commented:
@DelazJ  I missed your email DelazJ#99 and could not find it in my email. I need to check my personal settings. I was positive about the many places you have included, but consider it also as bold and brave, because it may cause Git merge conflicts to solve. I am not so brave so personally I would like to use more pull requests to achieve the same (if I could ;-) ).

I will later add some comments in line. (I have to run soon, literally!) I would appreciate if you would stick to the order 2D Map View, 3D map view, Elevation Profile view. 

Yes, you can add a 2D raster in an elevation profile, I did not manage to include a 2D vector and show the elevation in the elevation profile (but when it can be done, we certainly need to include it! ). Yes, we should include to describe the extrusion property that can be used to show a point as a vertical line (to indicate for instance the length of a pole or a diameter of a pipe (I need to test if it works for lines as well). In Deep Dive, Nyall Dawson explained how to use given expression to convert a 2D contour dataset to a 3D dataset to include it in the Elevation Profile. Therefore I wanted to include it, because it is certainly a solution to a problem people will face. Yes, I agree it can be included as a tip or a note!

Thanks for the invitation to keep discussing, polishing makes the end result shine

diethard2 commented on docs/user_manual/map_views/index.rst:
The original order is a more logical one. Although an elevation profile is maybe also a 2D view the concept is much harder to grasp and understand. A 3D is more natural for the enduser.

DelazJ replied:
I will revert, then.

diethard2 commented on docs/user_manual/map_views/elevation_profile.rst:
```diff
@@ -23,126 +25,203 @@ You can create several Elevation Profile views from the menu! When the first one
-   * When you have a DEM Raster (Digital Elevation Model) you can use the Processing function |processingAlgorithm| :guilabel:`Drape (set Z value from raster)` to add elevation to your vector layer.
-
-.. tip:: **Two great introduction video's on the Elevation Profile tool**
+Interacting with the profile plots
```
Replace profile plots with profile views. After adding the possibility to add an elevation profile in the plot layout (in version 3.30), it may otherwise cause confusion.

DelazJ replied:
OK, will turn this line to views to give us more freedom on description; we interact also with toolbar here. But I'm less sold to the confusion argument (please see below)

diethard2 commented on docs/user_manual/map_views/elevation_profile.rst:
```diff
@@ -23,126 +25,203 @@ You can create several Elevation Profile views from the menu! When the first one
+  and the :guilabel:`Total distance` between them, in map units.
+* |arrowLeft| :sup:`Nudge Left` and |arrowRight| :sup:`Nudge right` are used to shift
+  the position of the elevation profile line in the map canvas to its left or right.
+  The plot canvas will be redrawn, showing features and terrain overlapping the profile line buffer.
```
instead of plot canvas please use profile view

DelazJ replied:
The need is to name the area where the plot is rendered differently from the panel that opens when you hit the button in View menu. The latter is called every where "(elevation) profile view", and "plot canvas" is what Nyall used in his description for the former (e.g. #7485) and the way it is introduced at line 116 above. how about "profile canvas"?
I moreover don't think this will collide with layout; the item there is called "layout elevation profile item", and "plot layout" is not an used name afaict. OTOH what is rendered by that item in the layout is also what is displayed in the _profile view canvas_, so it is good imho they have a name, the same name. _and in this case I prefer plot over profile._

DelazJ replied:
While not exactly the same object, "chart area" used in the layout elevation item properties can also be an option?

diethard2 replied:
I agree when we first introduce the parts the Elevation Profile view:  A legend on the left,  a plot area on the right where the charts are drawn and on top a toolbar with options to create, and interact with, the elevation profile in the plot area.

DelazJ replied:
@diethard2 Am I correct that you agree with using "plot canvas"? And is there anything else we need to fix before merge? Thx.

diethard2 commented on docs/user_manual/map_views/elevation_profile.rst:
```diff
@@ -23,126 +25,203 @@ You can create several Elevation Profile views from the menu! When the first one
+  The plot canvas will be redrawn, showing features and terrain overlapping the profile line buffer.
+  The line is moved sideways using the :guilabel:`Tolerance` value in |options| :sup:`Options` menu.
+
+.. warning:: Closing an elevation profile view currently removes it from the project.
```
or closing the project

DelazJ commented:
> @DelazJ I missed your email https://github.com/DelazJ/QGIS-Documentation/pull/99 and could not find it in my email. I need to check my personal settings. I was positive about the many places you have included, but consider it also as bold and brave, because it may cause Git merge conflicts to solve. I am not so brave so personally I would like to use more pull requests to achieve the same (if I could ;-) )

I picked your email in KLIC viewer metadata (_I thought I was smart_ 😄) and get it confirmed by Mr Duivenvoorde. 
TBH conflict resolution with rst files (as long as you know what has changed 😉 ) is a real delight compared to same issue with .ui files. That one is a nightmare and if someone has a tip, I'm happy to learn.

DelazJ commented:
Let's merge and fix remaining issue(s) in next steps. If any, you can still report them in this thread @diethard2.
